src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java

```diff
@@ -112,10 +108,17 @@ private static void dispatchImpression(Imp imp, ExtImpAdkernel extImpAdkernel,
     }
 
     private static Imp compatImpression(Imp imp) {
-        final Imp.ImpBuilder impBuilder = imp.toBuilder().ext(null)
-                .audio(null)
-                .xNative(null);
-        return imp.getBanner() != null ? impBuilder.video(null).build() : impBuilder.build();
+        final Imp.ImpBuilder impBuilder = imp.toBuilder().ext(null).audio(null);
+
+        if (imp.getBanner() != null) {
+            return impBuilder.video(null).xNative(null).build();
+        } else if (imp.getVideo() != null) {
+            return impBuilder.banner(null).xNative(null).build();
+        } else if (imp.getXNative() != null) {
+            return impBuilder.banner(null).video(null).build();
```

Comment:
Due to if conditions higher banner and video already null, pls fix for 116 and 118 line